pkg/kopia/command/repository.go

```diff
@@ -88,6 +91,11 @@ func RepositoryCreateCommand(cmdArgs RepositoryCommandArgs) ([]string, error) {
 		args = args.AppendLoggableKV(overrideUsernameFlag, cmdArgs.Username)
 	}
 
+	if cmdArgs.RetentionMode != "" {
```

Comment:
Would it work if we just set `RetentionMode` and not the `RetentionPeriod`?
```suggestion
	if cmdArgs.RetentionMode != "" && cmdArgs.RetentionPeriod > 0 {
```

Reply:
@pavannd1 fyi I reverted this. kopia allows retention period to be 0 in 2 instances; when you want to disable retention options or when you want to change the mode but keep the period unchanged

https://github.com/kopia/kopia/blob/174f6141e133f487e54ba4cdede830636368aec2/cli/command_repository_set_parameters.go#L210-L218

If the mode is changed (but not disabled) and the period is 0 it will [return early](https://github.com/kopia/kopia/blob/174f6141e133f487e54ba4cdede830636368aec2/cli/command_repository_set_parameters.go#L105) and keep the existing value

Reply:
on creation side both are needed. RetentionPeriod should be >= 1 day so potentially could fail on this or just let it fail on kopia side

Reply:
Yeah, here it should just check if the values are available. Failure should come from kopia side